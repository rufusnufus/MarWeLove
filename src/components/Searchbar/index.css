.searchbar {
  width: 705px;
  height: 61px;
  background: white;
  border-radius: 30px;
  display: flex;
  margin: 150px auto 100px;
}

.searchbar__logo {
  height: 48px;
  margin: auto 0px auto 7px;
}

.searchbar__search {
  font-size: 22px;
  width: 100%;
  cursor: text;
  border-radius: 0 30px 30px 0;
  border: none;
}

textarea:focus,
input:focus {
  outline: none;
}
