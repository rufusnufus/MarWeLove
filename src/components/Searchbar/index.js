import './index.css';

function Searchbar() {
  return (
    <div className="searchbar">
        <img src="images/search.png" alt="" className="searchbar__logo" />
        <input type="text" placeholder="Search" className="searchbar__search" />
    </div>
  );
}

export default Searchbar;