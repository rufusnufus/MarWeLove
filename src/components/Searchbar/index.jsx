import React from 'react'

import './index.css'

function Searchbar({ onSubmit }) {
  const inputKeyUp = (e) => {
    if (e.key === 'Enter') {
      onSubmit(e.target.value)
    }
  }

  return (
    <div className="searchbar">
      <img src="images/search.png" alt="" className="searchbar__logo" />
      <input type="text" placeholder="Search" className="searchbar__search" onKeyUp={inputKeyUp} />
    </div>
  )
}

export default Searchbar
