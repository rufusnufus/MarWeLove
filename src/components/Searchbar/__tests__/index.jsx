/**
 * @jest-environment jsdom
 */

import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Searchbar from '../index'

afterEach(cleanup)

describe('SearchBar', () => {
  it('Render', () => {
    const { container } = render(<Searchbar onSubmit={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })
})
