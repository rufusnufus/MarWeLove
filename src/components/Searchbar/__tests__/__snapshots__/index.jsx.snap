// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Render 1`] = `
<div>
  <div
    class="searchbar"
  >
    <img
      alt=""
      class="searchbar__logo"
      src="images/search.png"
    />
    <input
      class="searchbar__search"
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
